src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -369,10 +369,10 @@ subroutine Infiltration(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, f
              else
                 qinmax=(1._r8 - fsat(c)) * minval(10._r8**(-e_ice*(icefrac(c,1:3)))*hksat(c,1:3))
              end if
-             qflx_infl_excess(c) = max(0._r8,qflx_in_soil(c) -  (1.0_r8 - frac_h2osfc(c))*qinmax)
+             qflx_infl_excess(c) = max(0._r8, qflx_in_soil(c) - qinmax) * (1.0_r8 - frac_h2osfc(c)) ! qflx_infl_excess only over the soil
 
              !4. soil infiltration and h2osfc "run-on"
-             qflx_infl(c) = qflx_in_soil(c) - qflx_infl_excess(c)
+             qflx_infl(c) = ( qflx_in_soil(c) - qflx_infl_excess(c) ) * (1.0_r8 - frac_h2osfc(c))  ! component over soil (move down?)
```

Comment:
**Comment from @martynpclark Aug 17, 2017**

The calculation of net fluxes here is tricky/confusing. The soil water input is a weighted combination of the influx over the non-ponded and ponded portion of the grid cell. Here we are just adding the non-ponded fraction.

It could be clearer if these calculations are moved down and done together.